plugin/evm/vm_test.go

```diff
@@ -810,6 +810,490 @@ func TestIssueAtomicTxs(t *testing.T) {
+	signedTx, err := types.SignTx(tx, types.NewEIP155Signer(vm.chainID), testKeys[0].ToECDSA())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	errs := vm.txPool.AddRemotesSync([]*types.Transaction{signedTx})
+	for _, err := range errs {
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	<-issuer
+
+	blk, err = vm.BuildBlock(context.Background())
+	if err != nil {
+		t.Fatal(err)
+	}
+	noAtomicTxBlock := blk.(*chain.BlockWrapper).Block.(*Block)
+
+	type testGetAtomicTx struct {
+		lastAcceptedBlockHeight uint64
+		preferredBlockHeight    uint64
+		blocks                  map[uint64]*types.Block
+		isApricotPhase5         map[uint64]bool
+		inputTxID               ids.ID
+		expectedTx              *Tx
+		expectedHeight          uint64
+		expectedErr             error
+	}
+
+	testCases := map[string]testGetAtomicTx{
+		"happy path": {
+			lastAcceptedBlockHeight: 1,
+			preferredBlockHeight:    2,
+			blocks:                  map[uint64]*types.Block{2: atomicTxBlock.ethBlock},
+			isApricotPhase5:         map[uint64]bool{2: false},
+			inputTxID:               importTx.ID(),
+			expectedTx:              importTx,
+			expectedHeight:          2,
+			expectedErr:             nil,
+		},
+		"no processing blocks to check": {
+			lastAcceptedBlockHeight: 2,
+			preferredBlockHeight:    2,
+			blocks:                  map[uint64]*types.Block{2: atomicTxBlock.ethBlock},
+			isApricotPhase5:         map[uint64]bool{2: false},
+			inputTxID:               importTx.ID(),
+			expectedTx:              nil,
+			expectedHeight:          0,
+			expectedErr:             errNoAtomicTxsFound,
+		},
+		"no atomic txs in processing blocks": {
+			lastAcceptedBlockHeight: 1,
+			preferredBlockHeight:    2,
+			blocks:                  map[uint64]*types.Block{2: noAtomicTxBlock.ethBlock},
+			isApricotPhase5:         map[uint64]bool{2: false},
+			inputTxID:               importTx.ID(),
+			expectedTx:              nil,
+			expectedHeight:          0,
+			expectedErr:             errNoAtomicTxsFound,
+		},
+		"search multiple processing blocks for atomic tx": {
+			lastAcceptedBlockHeight: 1,
+			preferredBlockHeight:    3,
+			blocks:                  map[uint64]*types.Block{2: noAtomicTxBlock.ethBlock, 3: atomicTxBlock.ethBlock},
+			isApricotPhase5:         map[uint64]bool{2: false, 3: false},
+			inputTxID:               importTx.ID(),
+			expectedTx:              importTx,
+			expectedHeight:          3,
+			expectedErr:             nil,
+		},
+	}
+
+	for _, test := range testCases {
+		tx, height, err := getAtomicTxFromProcessingBlocks(
+			test.inputTxID,
+			test.preferredBlockHeight,
+			test.lastAcceptedBlockHeight,
+			func(height uint64) bool { return test.isApricotPhase5[height] },
+			func(height uint64) *types.Block { return test.blocks[height] },
+			vm.codec)
+
+		if test.expectedErr != nil {
+			require.Equal(t, test.expectedErr, err)
+			continue
+		}
```

Comment:
this doesn't check `err == nil` in the case where `test.ExpectedErr == nil`.
Could we do:
```suggestion
		require.ErrorIs(t, test.expectedErr, err)
		if test.expectedErr != nil {
			require.Nil(t, tx)
			require.Zero(t, height)
			continue
		}
```

Reply:
thanks